File: Old as of September 2/old/emunana/emunana/ViewController.m
//
//  ViewController.m
//  emunana
//
//  Created by 若尾あすか on 2014/07/25.
//  Copyright (c) 2014年 若尾あすか. All rights reserved.
//
#import "ViewController.h"
#import <CoreMotion/CoreMotion.h>

@implementation ViewController



- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    [self updateStepCount];
}

// 歩数の表示
- (void)updateStepCount
{
    if([CMStepCounter isStepCountingAvailable])
    {
        // StepCounterのインスタンス作成
        CMStepCounter *stepCounter = [[CMStepCounter alloc] init];
        
        // 歩数を取得
        [stepCounter queryStepCountStartingFrom:[self startDateOfToday]
                                             to:[NSDate date]
                                        toQueue:[NSOperationQueue mainQueue]
                                    withHandler:^(NSInteger numberOfSteps, NSError *error) {
                                        self.stepLabel.text = [NSString stringWithFormat:@"%ld", numberOfSteps];
                                    }];
    }
}

// 今日の0時0分を取得
- (NSDate *)startDateOfToday
{
    NSCalendar *calendar = [NSCalendar currentCalendar];
    NSDateComponents *components = [calendar components:NSYearCalendarUnit|NSMonthCalendarUnit|NSDayCalendarUnit
                                               fromDate:[NSDate date]];
    
    return [calendar dateFromComponents:components];
}

@end

File: Old as of September 2/old/m7test/m7test/ViewController.swift
//
//  ViewController.swift
//  m7test
//
//  Created by ステファンアレクサンダー on 2014/07/28.
//  Copyright (c) 2014年 ステファンアレクサンダー. All rights reserved.
//

// Referenced http://stackoverflow.com/a/20938104 (Objective-C)

import UIKit
import CoreMotion

class ViewController: UIViewController {
    
    @IBOutlet weak var stepsCountingLabel: UILabel!
    var cmStepCounter:CMStepCounter!
    var operationQueue:NSOperationQueue!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        if (CMStepCounter.isStepCountingAvailable()) {
            cmStepCounter = CMStepCounter()
            cmStepCounter.startStepCountingUpdatesToQueue(operationQueue, updateOn: 1, withHandler: {(numberOfSteps:NSInteger, timeStamp:NSDate!, error:NSError!) in
                self.stepsCountingLabel.text = "\(numberOfSteps)"
            })
        }
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

